From: =?UTF-8?q?Guido=20G=C3=BCnther?= <agx@sigxcpu.org>
Date: Wed, 28 Jul 2010 09:37:32 +0200
Subject: [PATCH] Fix blacklist name

---
 udev/udev-add-printer |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/udev/udev-add-printer b/udev/udev-add-printer
index ce2b620..7189adc 100755
--- a/udev/udev-add-printer
+++ b/udev/udev-add-printer
@@ -27,7 +27,7 @@ import sys
 import traceback
 from syslog import *

-MFR_BLACKLIST=[
+MFG_BLACKLIST=[
     "graphtec",
     ]

--
