Index: system-config-printer-1.0.0/cupshelpers.py
===================================================================
--- system-config-printer-1.0.0.orig/cupshelpers.py	2008-05-27 11:42:33.000000000 +0200
+++ system-config-printer-1.0.0/cupshelpers.py	2008-06-21 11:18:18.770427107 +0200
@@ -19,7 +19,7 @@
 ## Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
 
 import cups, pprint, os, tempfile, re
-from rhpl.translate import _, N_
+from gettext import gettext as _
 import locale
 from debug import *
 
