--- a/openprinting.py	2008-02-17 12:44:10.544206173 +0100
+++ b/openprinting.py	2008-02-17 12:44:40.834358434 +0100
@@ -21,7 +21,10 @@
 
 import urllib, httplib, platform, threading, tempfile, traceback
 import os, sys
-from xml.etree.ElementTree import XML
+try:
+    from xml.etree.ElementTree import XML
+except ImportError:
+    from elementtree.ElementTree import XML
 
 def normalize_space (text):
     result = text.strip ()
