Index: system-config-printer-1.0.0/cupshelpers.py
===================================================================
--- system-config-printer-1.0.0.orig/cupshelpers.py	2008-09-12 22:52:08.243530986 +0200
+++ system-config-printer-1.0.0/cupshelpers.py	2008-09-12 22:53:03.556438702 +0200
@@ -208,6 +208,8 @@
         self.connection.setPrinterOpPolicy(self.name, policy)    
 
     def setJobSheets(self, start, end):
+        if not start: start = "none"
+        if not end: end = "none"
         self.connection.setPrinterJobSheets(self.name, start, end)
 
     def setAccess(self, allow, except_users):
