--- a/system-config-printer.glade	2008-06-08 13:11:01.229281585 +0200
+++ b/system-config-printer.glade	2008-06-08 13:13:11.829284169 +0200
@@ -139,7 +139,7 @@
 		      <child internal-child="image">
 			<widget class="GtkImage" id="image70">
 			  <property name="visible">True</property>
-			  <property name="stock">gtk-add</property>
+			  <property name="stock">gtk-new</property>
 			  <property name="icon_size">1</property>
 			  <property name="xalign">0.5</property>
 			  <property name="yalign">0.5</property>
