--- Makefile.am.orig	2008-06-08 13:19:54.097285829 +0200
+++ Makefile.am	2008-06-08 13:22:17.021810733 +0200
@@ -74,42 +74,38 @@
 install-desktopDATA: $(desktop_DATA)
 	mkdir -p $(DESTDIR)$(desktopdir)
 	mkdir -p $(DESTDIR)$(desktopdir)
-	desktop-file-install --vendor redhat	\
+	desktop-file-install	\
 	  --dir $(DESTDIR)$(desktopdir)		\
-	  --add-category X-Red-Hat-Base		\
 	  --add-category System			\
 	  --add-category Settings		\
 	  --add-category HardwareSettings	\
 	  --add-category Printing		\
 	  --add-category GTK			\
 	  system-config-printer.desktop
-	desktop-file-install --vendor redhat	\
+	desktop-file-install	\
 	  --dir $(DESTDIR)$(desktopdir)		\
-	  --add-category X-Red-Hat-Base		\
 	  --add-category System			\
 	  --add-category Monitor		\
 	  --add-category GTK			\
 	  manage-print-jobs.desktop
-	desktop-file-install --vendor redhat	\
+	desktop-file-install	\
 	  --dir $(DESTDIR)$(desktopdir)		\
-	  --add-category X-Red-Hat-Base		\
 	  --add-category Settings		\
 	  --add-category HardwareSettings	\
 	  --add-category GTK			\
 	  my-default-printer.desktop
-	desktop-file-install --vendor redhat	\
+	desktop-file-install	\
 	  --dir $(DESTDIR)$(autostartdir)	\
-	  --add-category X-Red-Hat-Base		\
 	  --add-category System			\
 	  --add-category Monitor		\
 	  --add-category GTK			\
 	  print-applet.desktop
 
 uninstall-desktopDATA:
-	rm -f $(DESTDIR)$(desktopdir)/redhat-system-config-printer.desktop
-	rm -f $(DESTDIR)$(desktopdir)/redhat-manage-print-jobs.desktop
-	rm -f $(DESTDIR)$(desktopdir)/redhat-my-default-printer.desktop
-	rm -f $(DESTDIR)$(autostartdir)/redhat-print-applet.desktop
+	rm -f $(DESTDIR)$(desktopdir)/system-config-printer.desktop
+	rm -f $(DESTDIR)$(desktopdir)/manage-print-jobs.desktop
+	rm -f $(DESTDIR)$(desktopdir)/my-default-printer.desktop
+	rm -f $(DESTDIR)$(autostartdir)/print-applet.desktop
 
 EXTRA_DIST=\
 	$(nobase_pkgdata_SCRIPTS) \
--- Makefile.in.orig	2008-05-27 19:14:22.000000000 +0200
+++ Makefile.in	2008-06-08 13:22:40.633281953 +0200
@@ -889,42 +889,38 @@
 install-desktopDATA: $(desktop_DATA)
 	mkdir -p $(DESTDIR)$(desktopdir)
 	mkdir -p $(DESTDIR)$(desktopdir)
-	desktop-file-install --vendor redhat	\
+	desktop-file-install	\
 	  --dir $(DESTDIR)$(desktopdir)		\
-	  --add-category X-Red-Hat-Base		\
 	  --add-category System			\
 	  --add-category Settings		\
 	  --add-category HardwareSettings	\
 	  --add-category Printing		\
 	  --add-category GTK			\
 	  system-config-printer.desktop
-	desktop-file-install --vendor redhat	\
+	desktop-file-install	\
 	  --dir $(DESTDIR)$(desktopdir)		\
-	  --add-category X-Red-Hat-Base		\
 	  --add-category System			\
 	  --add-category Monitor		\
 	  --add-category GTK			\
 	  manage-print-jobs.desktop
-	desktop-file-install --vendor redhat	\
+	desktop-file-install	\
 	  --dir $(DESTDIR)$(desktopdir)		\
-	  --add-category X-Red-Hat-Base		\
 	  --add-category Settings		\
 	  --add-category HardwareSettings	\
 	  --add-category GTK			\
 	  my-default-printer.desktop
-	desktop-file-install --vendor redhat	\
+	desktop-file-install	\
 	  --dir $(DESTDIR)$(autostartdir)	\
-	  --add-category X-Red-Hat-Base		\
 	  --add-category System			\
 	  --add-category Monitor		\
 	  --add-category GTK			\
 	  print-applet.desktop
 
 uninstall-desktopDATA:
-	rm -f $(DESTDIR)$(desktopdir)/redhat-system-config-printer.desktop
-	rm -f $(DESTDIR)$(desktopdir)/redhat-manage-print-jobs.desktop
-	rm -f $(DESTDIR)$(desktopdir)/redhat-my-default-printer.desktop
-	rm -f $(DESTDIR)$(autostartdir)/redhat-print-applet.desktop
+	rm -f $(DESTDIR)$(desktopdir)/system-config-printer.desktop
+	rm -f $(DESTDIR)$(desktopdir)/manage-print-jobs.desktop
+	rm -f $(DESTDIR)$(desktopdir)/my-default-printer.desktop
+	rm -f $(DESTDIR)$(autostartdir)/print-applet.desktop
 
 @INTLTOOL_DESKTOP_RULE@
 
