Index: system-config-printer-1.0.0/applet.py
===================================================================
--- system-config-printer-1.0.0.orig/applet.py	2008-06-22 16:37:24.966821412 +0200
+++ system-config-printer-1.0.0/applet.py	2008-06-22 16:45:27.922818962 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python
 
 ## Copyright (C) 2007, 2008 Tim Waugh <twaugh@redhat.com>
 ## Copyright (C) 2007, 2008 Red Hat, Inc.
@@ -19,9 +19,10 @@
 
 import cups
 import sys
+sys.path.append("/usr/share/system-config-printer")
 import statereason
 from statereason import StateReason
-from debug import *
+from cupsutils.debug import *
 import pprint
 
 import dbus
@@ -124,11 +125,11 @@ class NewPrinterNotification(dbus.servic
         os.unlink (filename)
         import sys
         sys.path.append (APPDIR)
-        import cupshelpers
+        from cupsutils import cupshelpers
         (missing_pkgs,
          missing_exes) = cupshelpers.missingPackagesAndExecutables (ppd)
 
-        from ppds import ppdMakeModelSplit
+        from cupsutils.ppds import ppdMakeModelSplit
         (make, model) = ppdMakeModelSplit (printer['printer-make-and-model'])
         driver = make + " " + model
         if status < self.STATUS_GENERIC_DRIVER:
Index: system-config-printer-1.0.0/authconn.py
===================================================================
--- system-config-printer-1.0.0.orig/authconn.py	2008-06-22 16:37:24.954820655 +0200
+++ system-config-printer-1.0.0/authconn.py	2008-06-22 16:44:51.635319642 +0200
@@ -19,7 +19,7 @@
 
 import cups
 import gtk
-from debug import *
+from cupsutils.debug import *
 
 _ = lambda x: x
 def set_gettext_function (fn):
Index: system-config-printer-1.0.0/contextmenu.py
===================================================================
--- system-config-printer-1.0.0.orig/contextmenu.py	2008-06-22 16:37:24.942820876 +0200
+++ system-config-printer-1.0.0/contextmenu.py	2008-06-22 16:44:51.639322665 +0200
@@ -23,7 +23,7 @@
 import cups
 import errordialogs
 import jobviewer
-from debug import *
+from cupsutils.debug import *
 
 _ = lambda x: x
 def set_gettext_function (x):
Index: system-config-printer-1.0.0/jobviewer.py
===================================================================
--- system-config-printer-1.0.0.orig/jobviewer.py	2008-06-22 16:37:24.934819719 +0200
+++ system-config-printer-1.0.0/jobviewer.py	2008-06-22 16:44:51.639322665 +0200
@@ -33,7 +33,7 @@ import pango
 import sys
 import time
 
-from debug import *
+from cupsutils.debug import *
 import config
 import statereason
 import errordialogs
Index: system-config-printer-1.0.0/monitor.py
===================================================================
--- system-config-printer-1.0.0.orig/monitor.py	2008-06-22 16:37:24.922820429 +0200
+++ system-config-printer-1.0.0/monitor.py	2008-06-22 16:44:51.639322665 +0200
@@ -22,7 +22,7 @@ import dbus
 import dbus.glib
 import gobject
 import time
-from debug import *
+from cupsutils.debug import *
 import pprint
 
 global _
Index: system-config-printer-1.0.0/pysmb.py
===================================================================
--- system-config-printer-1.0.0.orig/pysmb.py	2008-06-22 16:37:24.914820250 +0200
+++ system-config-printer-1.0.0/pysmb.py	2008-06-22 16:44:51.639322665 +0200
@@ -30,7 +30,7 @@ import gobject
 import gtk
 import os
 import pwd
-from debug import *
+from cupsutils.debug import *
 
 class AuthContext:
     def __init__ (self, parent=None, workgroup='', user='', passwd=''):
Index: system-config-printer-1.0.0/system-config-printer.py
===================================================================
--- system-config-printer-1.0.0.orig/system-config-printer.py	2008-06-22 16:37:24.902820959 +0200
+++ system-config-printer-1.0.0/system-config-printer.py	2008-06-22 16:45:44.063322922 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python
 
 ## system-config-printer
 
@@ -20,9 +20,6 @@
 ## along with this program; if not, write to the Free Software
 ## Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
 
-# config is generated from config.py.in by configure
-import config
-
 import errno
 import sys, os, tempfile, time, traceback, re, httplib
 import signal, thread
@@ -54,16 +51,21 @@ if len(sys.argv)>1 and sys.argv[1] == '-
 import cups
 cups.require ("1.9.27")
 
+sys.path.append("/usr/share/system-config-printer")
+# config is generated from config.py.in by configure
+import config
+
 import pysmb
-import cupshelpers, options
+from cupsutils import cupshelpers
+import options
 import gobject # for TYPE_STRING and TYPE_PYOBJECT
 from optionwidgets import OptionWidget
-from debug import *
-import ppds
+from cupsutils.debug import *
+from cupsutils import ppds
 import probe_printer
 import gtk_label_autowrap
 from gtk_treeviewtooltips import TreeViewTooltips
-import openprinting
+from cupsutils import openprinting
 import urllib
 import troubleshoot
 import contextmenu
Index: system-config-printer-1.0.0/troubleshoot/base.py
===================================================================
--- system-config-printer-1.0.0.orig/troubleshoot/base.py	2008-06-22 16:37:24.882820024 +0200
+++ system-config-printer-1.0.0/troubleshoot/base.py	2008-06-22 16:44:51.643323732 +0200
@@ -21,7 +21,7 @@
 
 import gtk
 from gettext import gettext as _
-from debug import *
+from cupsutils.debug import *
 
 TEXT_start_print_admin_tool = _("To start this tool, select "
                                 "System->Administration->Printing "
Index: system-config-printer-1.0.0/troubleshoot/CheckPPDSanity.py
===================================================================
--- system-config-printer-1.0.0.orig/troubleshoot/CheckPPDSanity.py	2008-06-22 16:37:24.874819845 +0200
+++ system-config-printer-1.0.0/troubleshoot/CheckPPDSanity.py	2008-06-22 16:44:51.643323732 +0200
@@ -20,7 +20,7 @@
 ## Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
 
 import cups
-import cupshelpers
+from cupsutils import cupshelpers
 import os
 import subprocess
 from base import *
Index: system-config-printer-1.0.0/my-default-printer.py
===================================================================
--- system-config-printer-1.0.0.orig/my-default-printer.py	2008-06-22 16:45:52.599322051 +0200
+++ system-config-printer-1.0.0/my-default-printer.py	2008-06-22 16:45:54.370814587 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python
 
 ## my-default-printer
 
