--- system-config-printer.glade.orig	2008-06-14 17:00:30.689968417 +0200
+++ system-config-printer.glade	2008-06-14 17:02:11.657969728 +0200
@@ -89,7 +89,7 @@
 		  <child>
 		    <widget class="GtkImageMenuItem" id="quit">
 		      <property name="visible">True</property>
-		      <property name="label">gtk-quit</property>
+		      <property name="label">gtk-close</property>
 		      <property name="use_stock">True</property>
 		      <signal name="activate" handler="on_quit_activate" last_modification_time="Thu, 23 Feb 2006 13:13:16 GMT"/>
 		    </widget>
