Index: system-config-printer-1.0.0/applet.py
===================================================================
--- system-config-printer-1.0.0.orig/applet.py	2008-06-21 11:18:30.729929293 +0200
+++ system-config-printer-1.0.0/applet.py	2008-06-21 11:18:32.917927838 +0200
@@ -264,19 +264,6 @@ if __name__ == '__main__':
         print >> sys.stderr, ("%s: unable to initialize pynotify" %
                               PROGRAM_NAME)
 
-    if trayicon:
-        # Stop running when the session ends.
-        def monitor_session (*args):
-            pass
-
-        try:
-            bus = dbus.SessionBus()
-            bus.add_signal_receiver (monitor_session)
-        except:
-            print >> sys.stderr, "%s: failed to connect to session D-Bus" % \
-                PROGRAM_NAME
-            sys.exit (1)
-
     try:
         bus = dbus.SystemBus()
     except:
