Index: system-config-printer-1.0.0/newprinternotification.conf
===================================================================
--- system-config-printer-1.0.0.orig/newprinternotification.conf	2008-06-21 11:21:02.050427218 +0200
+++ system-config-printer-1.0.0/newprinternotification.conf	2008-06-21 11:27:49.822964022 +0200
@@ -2,10 +2,6 @@
  "-//freedesktop//DTD D-BUS Bus Configuration 1.0//EN"
  "http://www.freedesktop.org/standards/dbus/1.0/busconfig.dtd">
 <busconfig>
-	<policy at_console="true">
-		<allow own="com.redhat.NewPrinterNotification"/>
-	</policy>
-
 	<policy user="root">
 		<allow own="com.redhat.NewPrinterNotification"/>
 	</policy>
@@ -16,7 +12,7 @@
 	</policy>
 	
         <policy context="default">
-                <deny own="com.redhat.NewPrinterNotification"/>
+                <allow own="com.redhat.NewPrinterNotification"/>
 
                 <deny send_destination="com.redhat.NewPrinterNotification"/>
                 <deny send_interface="com.redhat.NewPrinterNotification"/>
