Index: system-config-printer-1.0.0/newprinternotification.conf
===================================================================
--- system-config-printer-1.0.0.orig/newprinternotification.conf	2009-01-24 09:35:59.056653910 +0100
+++ system-config-printer-1.0.0/newprinternotification.conf	2009-01-24 09:36:11.272655106 +0100
@@ -8,13 +8,11 @@
 
 	<policy user="root">
 		<allow send_destination="com.redhat.NewPrinterNotification"/>
-                <allow send_interface="com.redhat.NewPrinterNotification"/>
 	</policy>
 	
         <policy context="default">
                 <allow own="com.redhat.NewPrinterNotification"/>
 
                 <deny send_destination="com.redhat.NewPrinterNotification"/>
-                <deny send_interface="com.redhat.NewPrinterNotification"/>
         </policy>
 </busconfig>
